File: FMAplication/ClientApp/src/app/Shared/Entity/Daily-Activity/posm-report.ts
import { DailyCMActivity } from '.';

export class PosmReport {

    public  id : number;
        public  dailyPOSMId : number;
        public  dailyCMActivityId : number;
        public  productId : number;
        public  productName : string;
        public  amount : number;
        public  actionType : number;
        public  productType : number;
        public  productPlanogramImageUrl : string;
        
        public  uploadedImageUrl1 : string;
       
        public  uploadedImageUrl2 : string;

        public  status : number;

        public dailyCMActivity: DailyCMActivity;


        //display

        public displayStatus: string;
        public displayActionType: string;


        public displayDate : string;
        public fmUserName : string;
        public cmUserName : string;
        public outletName : string;
        public salesPointName : string;

}

File: FMAplication/Domain/Reports/POSMReport.cs
using FMAplication.Core;
using FMAplication.Domain.DailyActivities;
using FMAplication.Domain.Products;
using FMAplication.Enumerations;
using System.ComponentModel.DataAnnotations;

namespace FMAplication.Domain.Reports
{
    public class POSMReport : AuditableEntity<int>
    {
        public int DailyPOSMId { get; set; }
        public int DailyCMActivityId { get; set; }
        public int ProductId { get; set; }
        public int Amount { get; set; }
        public POSMActionType ActionType { get; set; }
        public POSMProductType ProductType { get; set; }
        //[StringLength(512)]
        public string UploadedImageUrl1 { get; set; }
        //[StringLength(512)]
        public string UploadedImageUrl2 { get; set; }
        public POSMProduct Product { get; set; }

        public DailyCMActivity DailyCMActivity { get; set; }
    }
}
